web-ui/src/components/feedback_date_selector/SelectDate.jsx

```diff
@@ -18,75 +21,56 @@ pickerContain: {
-};
+    const sendDate = query.send && dateUtils.parse(query.send?.toString(), "yyyy-MM-dd");
+    const dueDate = query.due && dateUtils.parse(query.due?.toString(), "yyyy-MM-dd");
 
-const handleSendDateChange = (date) => {
-  setSendDateQuery(date.toString());
-  handleQueryChange("sendDate", date.toString())
-
-
-};
+    const handleDueDateChange = useCallback((date) => {
+        query.due = dateUtils.format(date, "yyyy-MM-dd");
+        history.push({...location, search: queryString.stringify(query)});
+    },[location, history, query]);
 
-return (
-<React.Fragment>
-  <div className = {classes.root}>
-  <div className={classes.pickerContain}>
-       <DatePicker
-       className= {classes.picker}
-               disableToolbar
-               format="MM/dd/yyyy"
-               margin="normal"
-               id="set-send-date"
-               label="Send Date:"
-                value={sendDateQuery}
-               onChange={handleSendDateChange}
-               KeyboardButtonProps={{
-                 'aria-label': 'change date',
-               }}
-             />
-            <DatePicker
-            className= {classes.picker}
-                disableToolbar
-                format="MM/dd/yyyy"
-                margin="normal"
-                id="set-due-date"
-                label="Due Date:"
-                emptyLabel="No due date"
-                value={dueDateQuery}
-                onChange={handleDueDateChange}
-                KeyboardButtonProps={{
-                   'aria-label': 'change date',
-                }}
-             />
-             </div>
-             </div>
-</React.Fragment>
+    const handleSendDateChange = useCallback((date) => {
+         query.send = dateUtils.format(date, "yyyy-MM-dd");
+        history.push({...location, search: queryString.stringify(query)});
+    },[location, history, query]);
 
-    );
+    return (
+    <React.Fragment>
+      <div className={classes.pickerContain}>
+           <DatePicker
+           className= {classes.picker}
+                   disableToolbar
+                   format="MM/dd/yyyy"
+                   margin="normal"
+                   id="set-send-date"
+                   label="Send Date:"
+                   value={sendDate}
+                   onChange={handleSendDateChange}
+                   KeyboardButtonProps={{
+                     'aria-label': 'change date',
+                   }}
+                 />
+                <DatePicker
+                className= {classes.picker}
+                    disableToolbar
+                    format="MM/dd/yyyy"
+                    margin="normal"
+                    id="set-due-date"
+                    label="Due Date:"
+                    emptyLabel="No due date"
+                    value={dueDate}
+                    onChange={handleDueDateChange}
+                    KeyboardButtonProps={{
+                       'aria-label': 'change date',
+                    }}
+                 />
```

Comment:
There is currently no way to set the due date as "No due date" although that's likely going to be a separate issue since it might require an additional button on the input field.

Reply:
Also, the date inputs are filled in by default but not included in the URL (so they are not used in the validation process). To most users, it will appear that the dates are valid and they should be able to proceed, but the next button is grayed out. It may be more intuitive to have those date fields blank by default. Or we could leave the default values and add them to the URL for validation as soon as they enter the date picker step.

Reply:
defaulting them to blank seems reasonable to me